src/nginx_config.py

```diff
@@ -3,344 +3,67 @@
 """Nginx workload."""
 
 import logging
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Set, cast
+from typing import Dict, List, cast
 
-import crossplane
 from charms.tempo_coordinator_k8s.v0.tracing import (
     ReceiverProtocol,
     TransportProtocolType,
     receiver_protocol_to_transport_protocol,
 )
-from cosl.coordinated_workers.coordinator import Coordinator
-from cosl.coordinated_workers.nginx import CERT_PATH, KEY_PATH, is_ipv6_enabled
+from cosl.coordinated_workers.nginx import (
+    NginxLocationConfig,
+)
 
 from tempo import Tempo
 from tempo_config import TempoRole
 
 logger = logging.getLogger(__name__)
-RESOLV_CONF_PATH = "/etc/resolv.conf"
 
 
-class NginxConfig:
+class NginxHelper:
```

Comment:
I'd get rid of this class altogether. all it has left is a couple of staticmethods: let's lift them to globals or toplevel functions and be done with it.